src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,130 @@
-import { useState } from 'react';
+import { FC, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type TNewMovie = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: FC<TNewMovie> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [inputFields, setInputFields] = useState({
+    title: '',
+    description: '',
+    imageUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    onAdd({
+      title: inputFields.title,
+      description: inputFields.description,
+      imgUrl: inputFields.imageUrl,
+      imdbUrl: inputFields.imdbUrl,
+      imdbId: inputFields.imdbId,
+    });
+
+    setInputFields({
+      title: '',
+      description: '',
+      imageUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+
+    setCount(prev => prev + 1);
+  };
+
+  const handleTitleChange = (currentValue: string) => {
+    setInputFields((prevState) => ({
+      ...prevState,
+      title: currentValue,
+    }));
+  };
+
+  const handleDescriptionChange = (currentValue: string) => {
+    setInputFields((prevState) => ({
+      ...prevState,
+      description: currentValue,
+    }));
+  };
+
+  const handleImageUrlChange = (currentValue: string) => {
+    setInputFields((prevState) => ({
+      ...prevState,
+      imageUrl: currentValue,
+    }));
+  };
+
+  const handleImdbUrlChange = (currentValue: string) => {
+    setInputFields((prevState) => ({
+      ...prevState,
+      imdbUrl: currentValue,
+    }));
+  };
+
+  const handleImdbIdChange = (currentValue: string) => {
+    setInputFields((prevState) => ({
+      ...prevState,
+      imdbId: currentValue,
+    }));
+  };
+
+  const isAllFieldValid = Object
+    .entries(inputFields)
+    .filter(([key]) => key !== 'description')
+    .every(([, val]) => val);
```

Comment:
You can just check values, you don't need keys here, something like this:
```suggestion
  const isAllFieldValid = Object
    .values(inputFields)
    .every((value) => value);
```